### DISCUSSION TITLE
Is the library thread-safe?

### DISCUSSION THREAD

**Discussion by gahisee** [spotify-web-api-java/spotify-web-api-java#368] [Q&A] [answered]
I just updated my 'old' program to the latest v8.3.3 from 2.x.x.
My app first makes the get-playlists request in one thread via ExecutorService.
Then, it immediately makes the get-devices request and it takes the exception:
**Connection [Not bound] is still allocated**

If I make the get-devices request when the get-playlists completes, all is fine.

**Comment by dargmuesli** [comment]
Are you using [SpotifyApiThreading's `executeAsync`](https://github.com/spotify-web-api-java/spotify-web-api-java/blob/476ad097036ec71fbf5dd55edc4e5276fb93ce29/src/main/java/se/michaelthelin/spotify/SpotifyApiThreading.java)? Would you share a minimal reproduction code example?

**Reply by gahisee** [reply, marked as answer]
Answer is in theory yes, but in practice regards to http connection, no. Since this library uses the BasicHttpClientConnectionManager which manages just one connection, it's best to simply make sync calls.

If I use PoolingHttpClientConnectionManager instead, the exception goes away. Just a few changes were made in SpotifyHttpManager as an experiment.

**Reply by dargmuesli** [reply]
Also see https://github.com/spotify-web-api-java/spotify-web-api-java/discussions/367#discussioncomment-7560730 which might help with the connection manager.
